File: Routes/area.py
from flask_restplus import Namespace, Resource
from Conversions import area

ns_area = Namespace('Area', description='Area calculations')


@ns_area.route('/Acre/<string:input>/ToSquareFoot')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.acre_to_ft2(input), 200
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/Acre/<string:input>/ToSquareMile')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.acre_to_mile2(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/Acre/<string:input>/ToSquareYard')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.acre_to_yard2(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/SquareFoot/<string:input>/ToAcre')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.ft2_to_acre(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/SquareFoot/<string:input>/ToSquareInch')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.ft2_to_in2(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/SquareInch/<string:input>/ToSquareFoot')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.in2_to_ft2(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/SquareInch/<string:input>/ToSquareYard')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.in2_to_yard2
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/SquareMile/<string:input>/ToAcre')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.mile2_to_acre(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/SquareRod/<string:input>/ToSquareYard')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.rod2_to_yard2(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/SquareYard/<string:input>/ToAcre')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.yard2_to_acre(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/SquareYard/<string:input>/ToSquareInch')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.yard2_to_in2(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/SquareYard/<string:input>/ToSquareRod')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.yard2_to_rod2(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400


@ns_area.route('/Circumference/<string:input>')
class ConversionsArea(Resource):
    def get(self, input):
        try:
            input = float(input)
            return area.circumference(input)
        except ValueError:
            return "Can't convert {} to number".format(input), 400

File: routes/area.js
const apiBasePath = '/api/v2/area/';

function acreToFoot2(input) {
  return input * 43560;
}

function acreToMile2(input) {
  return input * 0.0015625;
}

function acreToYard2(input) {
  return input * 4840;
}

function foot2ToAcre(input) {
  return input / 43560;
}

function foot2ToInch2(input) {
  return input * 144;
}

function inch2ToFoot2(input) {
  return input / 144;
}

function inch2ToYard2(input) {
  return input / 1296;
}

function mile2ToAcre(input) {
  return input * 640;
}

function rod2ToYard2(input) {
  return input * 30.25;
}

function yard2ToAcre(input) {
  return input / 4840;
}

function yard2ToInch2(input) {
  return input * 1296;
}

function yard2ToRod2(input) {
  return input * 0.0330578512396694;
}

function circumference(input) {
  return input * 3.14159265359;
}

export const areaRoutes = [
  {
    method: 'GET',
    path: `${apiBasePath}acreToFoot2/{input}`,
    handler: (request) => acreToFoot2(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}acreToMile2/{input}`,
    handler: (request) => acreToMile2(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}acreToYard2/{input}`,
    handler: (request) => acreToYard2(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}foot2ToAcre/{input}`,
    handler: (request) => foot2ToAcre(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}foot2ToInch2/{input}`,
    handler: (request) => foot2ToInch2(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}inch2ToFoot2/{input}`,
    handler: (request) => inch2ToFoot2(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}inch2ToYard2/{input}`,
    handler: (request) => inch2ToYard2(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}mile2ToAcre/{input}`,
    handler: (request) => mile2ToAcre(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}rod2ToYard2/{input}`,
    handler: (request) => rod2ToYard2(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}yard2ToAcre/{input}`,
    handler: (request) => yard2ToAcre(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}yard2ToInch2/{input}`,
    handler: (request) => yard2ToInch2(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}yard2ToRod2/{input}`,
    handler: (request) => yard2ToRod2(request.params.input)
  },
  {
    method: 'GET',
    path: `${apiBasePath}circumference/{input}`,
    handler: (request) => circumference(request.params.input)
  }
];
